torch/csrc/utils/pybind.h

```diff
@@ -6,19 +6,26 @@
 #include <pybind11/stl.h>
 
 #include "torch/csrc/DynamicTypes.h"
+#include "torch/csrc/autograd/python_variable.h"
 
 namespace py = pybind11;
 
 namespace pybind11 { namespace detail {
 
 // handle Tensor <-> at::Tensor conversions
+// Python Variables are unpacked into Tensors
 template <> struct type_caster<at::Tensor> {
 public:
   PYBIND11_TYPE_CASTER(at::Tensor, _("at::Tensor"));
 
   bool load(handle src, bool) {
+    PyObject* obj = src.ptr();
+    if (THPVariable_Check(obj)) {
+      value = ((THPVariable*)obj)->cdata.data();
```

Comment:
Should we assert that they don't require grad?

Reply:
The check would have to be (in pseudo-code): `!var.requires_grad() || !torch.is_grad_enabled()`

Not sure it's worth it. After we merge Tensor & Variable (but before release) we should revisit this. (Specifically the NCCL and comm functions)

Reply:
Fine, but make an issue for it or we will forget for sure

Reply:
Actually for now I'd just err to be on the safe side instead of silently ignoring `requires_grad`, which is completely different than what we're doing everywhere else